WordPress/Classes/Services/PostHelper.m

```diff
@@ -197,8 +198,17 @@ + (RemotePostCategory *)remoteCategoryWithCategory:(PostCategory *)category
     return remoteCategory;
 }
 
-+ (NSArray *)remoteMetadataForPost:(Post *)post {
-    NSMutableArray *metadata = [NSMutableArray arrayWithCapacity:4];
++ (NSArray *)remoteMetadataForPost:(Post *)post
+{
+    NSMutableArray *metadata = [NSMutableArray arrayWithCapacity:5];
+    
+    /// Send UUID as a foreign ID in metadata so we have a way to deduplicate new posts
+    if (!post.hasRemote && post.foreignID) {
```

Comment:
There is no need to check for `hasRemote` and it might break diffing.

```suggestion
    if (post.foreignID) {
```

If you simply include it in the metadata every time, the app will send it on the first `/create` but not on any of the subsequent edits automatically.

Reply:
Fixed [558ff58](https://github.com/wordpress-mobile/WordPress-iOS/pull/23173/commits/558ff5818e3bb082a36422e9e7652a96cbeafcc4)